### PR TITLE
Henrhoi/property-names-and-alias-change

### PR DESCRIPTION
**Pull request by henrhoi**
Have changed names on parameters, as discussed on Architect-meeting. I have only added the one alias "tracelevel".

I do not exactly see where I can and should add the new aliases, for the other parameters that do not have a QueryProfileType. Perhaps you could help me, or come with some tips? 

@bratseth

### CODE REVIEW

File: container-search/src/main/java/com/yahoo/prelude/searcher/FieldCollapsingSearcher.java [hunk 1]
```diff
@@ -29,8 +29,8 @@
 public class FieldCollapsingSearcher extends Searcher {
 
     private static final CompoundName collapse = new CompoundName("collapse");
-    private static final CompoundName collapsefield=new CompoundName("collapsefield");
-    private static final CompoundName collapsesize=new CompoundName("collapsesize");
+    private static final CompoundName collapsefield=new CompoundName("collapse.field");
+    private static final CompoundName collapsesize=new CompoundName("collapse.size");
```
Comment (bratseth) [inline container-search/src/main/java/com/yahoo/prelude/searcher/FieldCollapsingSearcher.java]: You'll need to add this to the query profile type in Query such that you can set the old names as aliases.

File: container-search/src/main/java/com/yahoo/prelude/semantics/SemanticSearcher.java [hunk 2]
```diff
@@ -29,7 +30,7 @@
 
     private static final CompoundName rulesRulebase=new CompoundName("rules.rulebase");
     private static final CompoundName rulesOff=new CompoundName("rules.off");
-    private static final CompoundName tracelevelRules=new CompoundName("tracelevel.rules");
+    private static final CompoundName tracelevelRules=new CompoundName("trace.rules");
```
Comment (bratseth) [inline container-search/src/main/java/com/yahoo/prelude/semantics/SemanticSearcher.java]: This is an API change. I think it's ok though, but there's probably something you need to change in the doc.

File: container-search/src/main/java/com/yahoo/search/Query.java [hunk 3]
```diff
@@ -198,7 +198,7 @@ public static Type getType(String typeString) {
         argumentType.addField(new FieldDescription(HITS.toString(), "integer", "hits count"));
         // TODO: Should this be added to com.yahoo.search.query.properties.QueryProperties? If not, why not?
         argumentType.addField(new FieldDescription(SEARCH_CHAIN.toString(), "string"));
-        argumentType.addField(new FieldDescription(TRACE_LEVEL.toString(), "integer", "tracelevel"));
+        argumentType.addField(new FieldDescription(TRACE_LEVEL.toString(), "integer", "trace.level tracelevel traceLevel"));
```
Comment (bratseth) [inline container-search/src/main/java/com/yahoo/search/Query.java]: I don't think this works? If the name has a dot it really means there's a nested type under "trace", and you need to add both levels (same as with model, ranking etc.).

Comment (bratseth) [inline container-search/src/main/java/com/yahoo/search/Query.java]: No need to add "trace.level" and "traceLevel" as aliases ...

**Review by bratseth — CHANGES_REQUESTED** [review]

File: container-search/src/main/java/com/yahoo/prelude/searcher/FieldCollapsingSearcher.java (hunk 1 above)
Comment (henrhoi) [reply container-search/src/main/java/com/yahoo/prelude/searcher/FieldCollapsingSearcher.java]: Can I make the CompoundNames public, so I can refer to them in Query, when adding them to the query profile type?

Comment (bratseth) [reply container-search/src/main/java/com/yahoo/prelude/searcher/FieldCollapsingSearcher.java]: Yes.

**Comment by bratseth** [comment]
Considering this more, I also think we should also wait with merging this until Vespa 7.
That will happen this quarter in any case. 
It won't break people using the search api over http but it will break any who set these parameters in an upstream searcher without using the static name variable (which they have not been able to in this case).

Anyway, it is good to complete it.

File: container-search/src/main/java/com/yahoo/search/Query.java (hunk 3 above)
Comment (henrhoi) [reply container-search/src/main/java/com/yahoo/search/Query.java]: Ah okay, but trace do not have it's own class like model, ranking etc. Should I make that, or 
create a QueryProfileType in Query, and with that adding new FieldDescriptions to the current QueryProfileType, like model, ranking etc?

I also see that trace.timestamps is a CompoundName, without a QueryProfileType in SearchHandler (ln 99). Should that also be added under "trace", as well as trace.rules in SemanticSearcher?

Comment (bratseth) [reply container-search/src/main/java/com/yahoo/search/Query.java]: I think it's good enough to add the QueryProfileType and no separate class.
Yes, trace.timestamps should be under that as well then.

Comment (henrhoi) [reply container-search/src/main/java/com/yahoo/search/Query.java]: Tried to do this in latest commit.

**Pull request closed without merging**